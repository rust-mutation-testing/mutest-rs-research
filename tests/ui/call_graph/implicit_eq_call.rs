//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty

struct ImplsEq;

fn eq_impl() {}
fn ne_impl() {}

impl PartialEq for ImplsEq {
    fn eq(&self, _other: &Self) -> bool {
        eq_impl();
        true
    }

    fn ne(&self, _other: &Self) -> bool {
        ne_impl();
        false
    }
}

impl Eq for ImplsEq {}

#[test]
fn test_implicit_eq_call() {
    let _ = ImplsEq == ImplsEq;
}

#[test]
fn test_implicit_ne_call() {
    let _ = ImplsEq != ImplsEq;
}
