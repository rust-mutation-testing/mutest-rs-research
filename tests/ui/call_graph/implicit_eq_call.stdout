
@@@ call graph @@@

entry points:

test test_implicit_eq_call
  -> <ImplsEq as std::cmp::PartialEq>::eq at tests/ui/call_graph/implicit_eq_call.rs:12:5: 12:40 (#0)
test test_implicit_ne_call
  -> <ImplsEq as std::cmp::PartialEq>::ne at tests/ui/call_graph/implicit_eq_call.rs:17:5: 17:40 (#0)

nested calls at distance 1:

<ImplsEq as std::cmp::PartialEq>::eq at tests/ui/call_graph/implicit_eq_call.rs:12:5: 12:40 (#0)
  -> eq_impl at tests/ui/call_graph/implicit_eq_call.rs:8:1: 8:13 (#0)
<ImplsEq as std::cmp::PartialEq>::ne at tests/ui/call_graph/implicit_eq_call.rs:17:5: 17:40 (#0)
  -> ne_impl at tests/ui/call_graph/implicit_eq_call.rs:9:1: 9:13 (#0)

@@@ targets @@@

tests -(1)-> eq_impl at tests/ui/call_graph/implicit_eq_call.rs:8:1: 8:13 (#0)
  (1) test_implicit_eq_call

tests -(1)-> ne_impl at tests/ui/call_graph/implicit_eq_call.rs:9:1: 9:13 (#0)
  (1) test_implicit_ne_call

tests -(0)-> <ImplsEq as std::cmp::PartialEq>::eq at tests/ui/call_graph/implicit_eq_call.rs:12:5: 12:40 (#0)
  (0) test_implicit_eq_call

tests -(0)-> <ImplsEq as std::cmp::PartialEq>::ne at tests/ui/call_graph/implicit_eq_call.rs:17:5: 17:40 (#0)
  (0) test_implicit_ne_call

targets: 4 total; 4 safe; 0 unsafe (0 tainted)
